### PR TITLE
Handle sincos

### PR DESCRIPTION
**Pull request by wsmoses**
Should fix both https://github.com/wsmoses/Enzyme.jl/issues/219 and https://github.com/wsmoses/Enzyme.jl/issues/99

### CODE REVIEW

File: src/compiler/interpreter.jl [hunk 1]
```diff
@@ -70,11 +70,13 @@ end
 function is_primitive_func(@nospecialize(TT))
     isa(TT, DataType) || return false
     ft = TT.parameters[1]
+    if ft === typeof(Base.string)
```
Comment (vchuravy) [inline src/compiler/interpreter.jl:73]: This was intentionally only for `string` with FloatingPoint args

**Review by vchuravy — APPROVED** [review]

File: src/compiler/interpreter.jl (hunk 1 above)
Comment (wsmoses) [reply src/compiler/interpreter.jl:73]: I mean regardless this is inactive and we want to mark as such?

**Comment by wsmoses** [comment]
@vchuravy the remaining failure appears to be unrelated due to a change in SpecialFunctions.jl

Good to merge?

File: src/compiler/interpreter.jl (hunk 1 above)
Comment (vchuravy) [reply src/compiler/interpreter.jl:73]: Yes but we don't want to change inlining behavior more than necessary

**Comment by vchuravy** [comment]
Reproducer:

```
using Enzyme
using SpecialFunctions

Enzyme.autodiff((y) -> SpecialFunctions.bessely(2, y), Active(1))
```

**Pull request merged**